partition_snapshot_row_cursor.hh

```diff
@@ -401,21 +538,43 @@ public:
     // The cursor remains valid after the call and points at the same row as before.
     ensure_result ensure_entry_in_latest() {
         auto&& rows = _snp.version()->partition().mutable_clustered_rows();
-        auto latest_i = get_iterator_in_latest_version();
-        rows_entry& latest = *latest_i;
         if (is_in_latest_version()) {
+            auto latest_i = get_iterator_in_latest_version();
+            rows_entry& latest = *latest_i;
             if (_snp.at_latest_version()) {
                 _snp.tracker()->touch(latest);
             }
             return {latest, false};
         } else {
             // Copy row from older version because rows in evictable versions must
             // hold values which are independently complete to be consistent on eviction.
-            auto e = current_allocator().construct<rows_entry>(_schema, *_current_row[0].it);
-            e->set_continuous(latest_i && latest_i->continuous());
-            _snp.tracker()->insert(*e);
-            rows.insert_before(latest_i, *e);
-            return {*e, true};
+            auto e = [&] {
+                if (!at_a_row()) {
+                    return alloc_strategy_unique_ptr<rows_entry>(
```

Comment:
What does this alloc_strategy_unique_ptr do and why is it needed in the new version and not needed in the previous version?

Reply:
It's like std::unique_ptr but invokes current_allocator() to delete stuff (so can be used with LSA-managed objects)

I need it because I call a throwing insert() below in the reverse path, and we need to free the object on exception from insert().

Turns out that before we also needed it, because insert_before() is throwing too, @xemul fixed it recently.